src/Behat/Behat/Definition/Cli/UnusedDefinitionsController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Definition\Cli;
+
+use Behat\Behat\Definition\Call\RuntimeDefinition;
+use Behat\Behat\Definition\DefinitionRepository;
+use Behat\Behat\Definition\Printer\ConsoleDefinitionInformationPrinter;
+use Behat\Testwork\Cli\Controller;
+use Behat\Testwork\EventDispatcher\Event\AfterSuiteTested;
+use Behat\Testwork\EventDispatcher\Event\ExerciseCompleted;
+use Behat\Testwork\EventDispatcher\Event\SuiteTested;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Prints unused definitions
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+final class UnusedDefinitionsController implements Controller
+{
+    /**
+     * @var array <string, array{definition: RuntimeDefinition, used: bool}>
+     */
+    private array $definitionUsage = [];
+
+    public function __construct(
+        private DefinitionRepository $definitionRepository,
+        private EventDispatcherInterface $eventDispatcher,
+        private ConsoleDefinitionInformationPrinter $printer,
+    ) {
+    }
+
+    public function configure(Command $command): void
+    {
+        $command
+            ->addOption(
+                '--print-unused-definitions', null, InputOption::VALUE_NONE,
+                "Reports definitions that were never used."
+            )
+        ;
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        if ($input->getOption('print-unused-definitions')) {
+            $this->eventDispatcher->addListener(SuiteTested::AFTER, array($this, 'registerDefinitionUsages'), -999);
+            $this->eventDispatcher->addListener(ExerciseCompleted::AFTER, array($this, 'printUnusedDefinitions'), -999);
+        }
+        return null;
+    }
+
+    public function registerDefinitionUsages(AfterSuiteTested $event): void
+    {
+        $environmentDefinitions = $this->definitionRepository->getEnvironmentDefinitions($event->getEnvironment());
```

Comment:
This listener is called after each suite is tested. We get a list of all found definitions and save them in an array, noting if they have been used or not